pkg/controller/atlasproject/team_reconciler.go

```diff
@@ -132,53 +133,49 @@ func (r *AtlasProjectReconciler) teamReconcile(
 	}
 }
 
-func ensureTeamState(workflowCtx *workflow.Context, team *akov2.AtlasTeam) (string, workflow.Result) {
-	var atlasTeamResponse *admin.TeamResponse
+func (r *AtlasProjectReconciler) ensureTeamState(workflowCtx *workflow.Context, team *akov2.AtlasTeam) (string, workflow.Result) {
+	var atlasTeam *teams.Team
 	var err error
 
 	if team.Status.ID != "" {
-		atlasTeamResponse, err = fetchTeamByID(workflowCtx, team.Status.ID)
-		if err != nil {
-			return "", workflow.Terminate(workflow.TeamNotCreatedInAtlas, err.Error())
-		}
+		atlasTeam, err = r.fetchTeamByID(workflowCtx, team.Status.ID)
+		if err == nil {
+			atlasTeam, err = r.renameTeam(workflowCtx, atlasTeam, team.Spec.Name)
+			if err != nil {
+				return "", workflow.Terminate(workflow.TeamNotUpdatedInAtlas, err.Error())
+			}
 
-		atlasTeamResponse, err = renameTeam(workflowCtx, atlasTeamResponse, team.Spec.Name)
+			return atlasTeam.TeamID, workflow.OK()
+		}
+	} else {
+		atlasTeam, err = r.fetchTeamByName(workflowCtx, team.Spec.Name)
 		if err != nil {
-			return "", workflow.Terminate(workflow.TeamNotUpdatedInAtlas, err.Error())
+			return "", workflow.Terminate(workflow.TeamNotCreatedInAtlas, err.Error())
 		}
-
-		return atlasTeamResponse.GetId(), workflow.OK()
-	}
-
-	atlasTeamResponse, err = fetchTeamByName(workflowCtx, team.Spec.Name)
-	if err != nil {
-		return "", workflow.Terminate(workflow.TeamNotCreatedInAtlas, err.Error())
 	}
 
-	if atlasTeamResponse == nil {
-		atlasTeam, err := team.ToAtlas()
-		if err != nil {
+	if atlasTeam == nil {
+		desiredAtlasTeam := teams.NewAssignedTeam(&team.Spec, team.Status.ID)
+		if desiredAtlasTeam == nil {
```

Comment:
this won't work. You don't have an `err` in the first place for the `workflow.Terminate`.